Fix implicit declaration of ncurses wide-character functions.

I don't know whether FreeBSD should be handled the same way as macOS.
For MacPorts, only the macOS case is relevant.
--- src/clexheaders.h.orig	2009-01-11 13:44:35.000000000 -0600
+++ src/clexheaders.h	2022-07-21 08:36:41.000000000 -0500
@@ -1,4 +1,6 @@
-#if !defined(__APPLE__) && !defined(__FreeBSD__)
+#if defined(__APPLE__)
+# define _XOPEN_SOURCE_EXTENDED
+#elif !defined(__FreeBSD__)
 # define _XOPEN_SOURCE 600
 #endif
 
--- src/kbd-test.c.orig	2020-09-26 11:08:38.000000000 -0500
+++ src/kbd-test.c	2022-07-21 08:41:39.000000000 -0500
@@ -14,6 +14,8 @@
 
 #include "../config.h"
 
+#include "clexheaders.h"
+
 #include <locale.h>
 #include <stdlib.h>
 #include <time.h>
